labgrid/driver/networkusbstoragedriver.py

```diff
@@ -46,6 +50,53 @@ def on_activate(self):
     def on_deactivate(self):
         pass
 
+    def _pmount(self, dev_path, mount_path):
+        mount_args = ["pmount", dev_path]
+        self.logger.debug('Mount {} to {}'.format(dev_path, mount_path))
+        processwrapper.check_output(self.storage.command_prefix + mount_args)
+
+    def _pumount(self, dev_path):
+        for i in range(self.PUMOUNT_MAX_RETRIES):
+            sync_args = ["sync", dev_path]
+            processwrapper.check_output(
+                self.storage.command_prefix + sync_args)
+
+            umount_args = ["pumount", dev_path]
+            try:
+                processwrapper.check_output(
+                    self.storage.command_prefix + umount_args)
+            except subprocess.CalledProcessError as e:
+                if e.returncode == 5:
+                    self.logger.info(
+                            'umount: {}: target is busy; wait for {} s'.format(
+                                dev_path, self.PUMOUNT_BUSY_WAIT))
+                    time.sleep(self.PUMOUNT_BUSY_WAIT)
+                    continue
+                else:
+                    raise e
```

Comment:
Wouldn't it be better to do something similar to `NetworkUSBStorageDriver.write_image()`'s waiting for medium? Maybe move that to a separate method and use it? I guess you should wait for the partition size to be >0 there.

Reply:
I tried this out, but the value of the size sysfs attribute does not seem to have a causal relationship to the partition being ready to be unmounted. Do you have any documentation for this "trick"?

Reply:
Are you sure? This works for me:
```
labgrid-client sd-mux dut
(while [ true ]; do [ "$(cat /sys/class/block/sdc1/size 2>/dev/null || echo 0)" -gt 0 ] && break; done; pmount /dev/sdc1)&
labgrid-client sd-mux host
ls /media/sdc1
```

Reply:
Ok, but this is about **mounting** now, the comment refers to lines 59-76 though, which deal with **unmounting**.

Reply:
You are right, I misread that. You should then implement that for the mount case.

Reply:
Done, please review.